playbooks/common/openshift-master/config.yml

```diff
@@ -235,36 +235,58 @@
   - role: haproxy
     when: groups.oo_masters_to_config | length > 1
 
-- name: Generate master session keys
+- name: Determine if session secrets need to be generated
   hosts: oo_first_master
-  tasks:
+  pre_tasks:
   - fail:
       msg: "Both openshift_master_session_auth_secrets and openshift_master_session_encryption_secrets must be provided if either variable is set"
     when: (openshift_master_session_auth_secrets is defined and openshift_master_session_encryption_secrets is not defined) or (openshift_master_session_encryption_secrets is defined and openshift_master_session_auth_secrets is not defined)
   - fail:
       msg: "openshift_master_session_auth_secrets and openshift_master_encryption_secrets must be equal length"
     when: (openshift_master_session_auth_secrets is defined and openshift_master_session_encryption_secrets is defined) and (openshift_master_session_auth_secrets | length != openshift_master_session_encryption_secrets | length)
+  roles:
+  - role: openshift_facts
+  post_tasks:
+  - openshift_facts:
+      role: master
+      local_facts:
+          session_auth_secrets: "{{ openshift_master_session_auth_secrets | default(openshift.master.session_auth_secrets | default(None)) }}"
+          session_encryption_secrets: "{{ openshift_master_session_encryption_secrets | default(openshift.master.session_encryption_secrets | default(None)) }}"
+
+- name: Generate master session keys
+  hosts: oo_first_master
+  pre_tasks:
+  - set_fact:
+      session_secrets_missing: "{{ not (openshift.master.session_auth_secrets | default(false) or openshift.master.session_encryption_secrets | default(false)) }}"
   - name: Install OpenSSL package
     action: "{{ansible_pkg_mgr}} pkg=openssl state=present"
   - name: Generate session authentication key
     command: /usr/bin/openssl rand -base64 24
     register: session_auth_output
     with_sequence: count=1
-    when: openshift_master_session_auth_secrets is undefined
+    when: session_secrets_missing | bool
```

Comment:
Instead of using command, register and set_fact later, we should probably move this to a lookup plugin or filter plugin. Then we could also just use the openssl python lib then instead of execing out to openssl.